### PR TITLE
Disable clang format in the plugin registrants

### PR DESCRIPTION
**Pull request by blasten**
As per https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code

### CODE REVIEW

File: packages/flutter_tools/lib/src/plugins.dart
Comment (stuartmorgan) [inline packages/flutter_tools/lib/src/plugins.dart:1234]: clang format shouldn't be running on CMake files.

**Review by stuartmorgan-g — APPROVED** [review]
Hm, interesting. I was thinking we would turn this off via filtering in the flutter/plugins autoformat tooling.

On one hand, it feels odd to make declarations about how the clients should (or in this case, shouldn't) format the generated code, but on the other hand I can't think of likely scenarios where someone would actually want to autoformat code that's going to be stomped on every build.

I guess I come down on the side of LGTM (modulo CMake), but let's get a second opinion.

**Review by jmagman — APPROVED** [review]
Other than @stuartmorgan's cmake comment, if it works, LGTM!

File: packages/flutter_tools/lib/src/plugins.dart
Comment (blasten) [reply packages/flutter_tools/lib/src/plugins.dart:1234]: ahh!

**Comment by Hixie** [comment]
This change means that running the tests locally changes your local workspace.
I've applied those changes (I think) in https://github.com/flutter/flutter/pull/77093

**Comment by stuartmorgan** [comment]
This was apparently reverted at some point. Can we re-land it? what was the reason for the revert?

**Comment by Hixie** [comment]
cc @zanderso who made the reverts (directly to the tree?)

**Comment by zanderso** [comment]
A series of commits had to be reverted here https://github.com/flutter/flutter/pull/78623, to avoid the regressions described in 
https://github.com/flutter/flutter/issues/78554 from appearing in a beta release.

**Comment by stuartmorgan** [comment]
This PR wasn't related to that at all, which is why I was confused as to why it was reverted. I guess it just got caught in the net by accident.

**Comment by zanderso** [comment]
Yeah, sorry about not broadcasting the reverts more widely. I just cc'd @blasten there. I believe this change did conflict with a revert that was related, so it unfortunately got swept in with the others.

**Pull request merged**
